extensions/amp-analytics/0.1/requests.js

```diff
@@ -428,3 +471,33 @@ function expandExtraUrlParams(
 
   return Promise.all(requestPromises).then(() => params);
 }
+
+/**
+ * Composes a request URL given a base and requestOrigin
+ * @private
+ * @param {string} baseUrl
+ * @param {string=} opt_requestOrigin
+ * @return {string}
+ */
+function composeRequestUrl_(baseUrl, opt_requestOrigin) {
+  if (opt_requestOrigin === '') {
+    user().error(TAG, 'Request Origin cannot be an empty string');
+    return '';
+  }
+  // prepend requestOrigin if available
+  else if (opt_requestOrigin) {
+    const domain = extractDomain(opt_requestOrigin);
+
+    if (!domain) {
+      user().error(
+        TAG,
+        'Unable to extract domain from requestOrigin: ' + opt_requestOrigin
+      );
+      return '';
+    }
+
+    return domain + baseUrl;
```

Comment:
Do we always require leading `/` in baseUrl?  Should we append one if it doesn't exist?

Reply:
baseUrl will already be encoded when it is passed in.

I don't think we should append a `/` if one doesn't exist because the user may want a `?` or `#` etc. instead. I think it will be simpler for the user to handle that themselves.